lnrpc/walletrpc/walletkit_server.go

```diff
@@ -645,6 +649,67 @@ func (w *WalletKit) PublishTransaction(ctx context.Context,
 	return &PublishResponse{}, nil
 }
 
+// RemoveTransaction attempts to remove the transaction and all of its
+// descendants resulting from further spends of the outputs of the provided
+// transaction id.
+// NOTE: We do not remove the transaction from the rebroadcaster which might
+// run in the background rebroadcasting not yet confirmed transactions. We do
+// not have access to the rebroadcaster here nor should we. This command is not
+// a way to remove transactions from the network. It is a way to shortcircuit
+// wallet utxo housekeeping while transactions are still unconfirmed and we know
+// that a transaction will never confirm because a replacement already pays
+// higher fees.
+func (w *WalletKit) RemoveTransaction(_ context.Context,
+	req *TransactionID) (*RemoveTransactionResponse, error) {
+
+	// Convert the byte slice into the `Hash` type to make sure the txid has
+	// a valid format.
+	txHash, err := chainhash.NewHash(req.Txid)
+	if err != nil {
+		return nil, err
+	}
+
+	// Query the tx store of our internal wallet for unconfirmed
+	// transactions only. The accountFilter is currently not used in the
+	// `ListTransactionDetails` function so we provide an empty string.
+	// -1 equals the mempool height so only unconfirmed transactions are
+	// returned.
+	transactions, err := w.cfg.Wallet.ListTransactionDetails(-1, -1, "")
```

Comment:
We can now fetch a transaction directly from the wallet, see https://github.com/lightningnetwork/lnd/pull/7654.

Reply:
I wonder if we also should allow to remove confirmed transaction to shortcircuit the wallet rescan, but I guess for this we would also need to introduce a `addTx` cmd, so that that we can manually add some utxos if we know for sure that they are ours?

Reply:
see for example issue #8251, where they kind of know what they are missing ?

Reply:
Yeah, not sure if that's a good idea... You could really mess up your wallet state with that. At the very least it shouldn't be part of this PR anyway so we can properly discuss the feature (and its consequences) in a standalone PR.